src/dotnet/commands/dotnet-nuget/NuGetCommand.cs

```diff
@@ -0,0 +1,83 @@
+// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Reflection;
+using Microsoft.DotNet.Cli.CommandLine;
+using Microsoft.DotNet.Cli.Utils;
+using Microsoft.DotNet.InternalAbstractions;
+using NugetProgram = NuGet.CommandLine.XPlat.Program;
+
+namespace Microsoft.DotNet.Tools.NuGet
+{
+    public class NuGetCommand
+    {
+        public static int Run(string[] args)
+        {
+            DebugHelper.HandleDebugSwitch(ref args);
+
+            var app = new CommandLineApplication(false)
+            {
+                Name = "dotnet nuget",
+                FullName = ".NET NuGet command runner",
+                Description = "For running locals, push and delete commands (\"dotnet nuget [command] --help\" for specifics)"
+            };
+
+
+            app.OnExecute(() =>
+            {
+                try
+                {
+                    return RunCommand(args, new NuGetCommandRunner());
+                }
+                catch (InvalidOperationException e)
+                {
+                    Console.WriteLine(e.Message);
+
+                    return -1;
+                }
+                catch (Exception e)
+                {
+                    Console.WriteLine(e.Message);
+
+                    return -2;
+                }
+            });
+
+            return app.Execute(args);
+        }
+
+        public static int RunCommand(IEnumerable<string> args, INuGetCommandRunner commandRunner)
+        {
+            Debug.Assert(commandRunner != null, "A command runner must be passed to RunCommand");
+            if (commandRunner == null)
+            {
+                return -1; // invalid operation
+            }
+
+            string[] commandArgs = args.ToArray();
+            if (commandArgs.Length == 0 ||
+                !commandArgs[0].Equals("locals", StringComparison.OrdinalIgnoreCase) &&
+                !commandArgs[0].Equals("push", StringComparison.OrdinalIgnoreCase) &&
+                !commandArgs[0].Equals("delete", StringComparison.OrdinalIgnoreCase))
+            {
+                return 1; // unsupported argument
+            }
+
+            return commandRunner.Run(commandArgs);
+        }
+
+        private class NuGetCommandRunner : INuGetCommandRunner
+        {
+            public int Run(string[] commandArgs)
+            {
+                var nugetAsm = typeof(NugetProgram).GetTypeInfo().Assembly;
```

Comment:
@rohit21agrawal is working on updating the pack command. I believe the plan at this point is to keep it as dotnet pack and not dotnet nuget pack, although the cli help on this point is misleading. I agree with the idea of making it consistent, but we'd like to avoid breaking everybody. If the two operate as dumb passthroughs to the nuget dll, perhaps code could be shared, and both dotnet pack and dotnet nuget pack would light up. Unsure. @harikmenon may have thoughts on this too.